### PR TITLE
TD-2652 TreeViewList component - improvements and bugs

### PR DESCRIPTION
**Pull request by syedsalehinipg**
Closes [TD-2652](https://sce.myjetbrains.com/youtrack/issue/TD-2652/TreeViewList-component-improvements-and-bugs)

## Changes

This PR addresses the following issues:

1. After a parameter is selected and then the top level component is collapsed, the selected parameter parameter was losing the selection
      - Current selection state has been added to keep the selection active when opening/closing parent nodes
2. The selected child node was not covering the entire width.
      - Extra padding removed
      - Override the `.css-9l5vo-MuiCollapse-wrapperInne` class which had the `width` set to `"100%"` this was causing the issue of not covering the entire width, setting it to `auto` fixed the issue
3. There was a flicker in the UI when hovering over the uncovered area in the child nodes
      - As the flickering was only happening in the uncovered area this is now fixed as there is no covered area

Other:

- Simplified use of `props.hoveredNode` with `hoveredNode`
- Updated example items json to include a new item with longer label and removed spaces in labels
- Updated tests to match new labels
- Small change: `TabPanel.jsx` - removed padding for child tabs

**Issues:**

_The selection or hover is not covering the entire width._

image

_Flicker_

https://github.com/IPG-Automotive-UK/react-ui/assets/143519265/e04440a9-c0a8-40dd-bc64-c1cf382e874a

## UI/UX

_Selection covers full width including the scrollable area_
image

## Testing notes

Navigate to the **Custom Width** story for **TreeViewList** component, using the `width` props and set to `200px`

Tests:

1. Select a child node, collapse the parent node, and then expand again - the selected child node should remain highlighted
2. Expand **AER**, select **ConsiderationPointPosition**, the highlighted selection should cover the entire width, check by scrolling horizontally
3. Hover over expanded items, no flickering should take place

## Author checklist before assigning a reviewer

- [x] Reviewed my own code-diff.
~~- [ ] Branch has been run in docker.~~
- [x] PR assigned to me or an appropriate delegate.
- [x] Relevant labels added to the PR.
~~- [ ] Appropriate tests have been added.~~
- [x] Lint and test workflows pass.

### CODE REVIEW

File: src/TreeViewList/example-items.json [hunk 1]
```diff
@@ -5,18 +5,23 @@
     "tooltip": "Aerodynamics description goes here",
```
Comment (AronPetrauskas) [inline src/TreeViewList/example-items.json:5]: I assume this tooltip is a placeholder and needs to be replaced.

File: src/TreeViewList/TreeViewList.spec.tsx [hunk 2]
```diff
@@ -36,7 +36,7 @@ test("should display parameter value once expanded", async ({ page }) => {
   await expect(
     page
       .frameLocator('iframe[title="storybook-preview-iframe"]')
-      .getByText("Drag Coefficient", { exact: true })
+      .getByText("DragCoefficient", { exact: true })
```
[the remark this answers is not in the document]
Comment (syedsalehinipg) [reply src/TreeViewList/TreeViewList.spec.tsx:39]: To match current uses in VIRTO.DATA, and stop line wrapping

File: src/TreeViewList/example-items.json (hunk 1 above)
Comment (syedsalehinipg) [reply src/TreeViewList/example-items.json:5]: This is just an example

**Comment by syedsalehinipg** [comment]
The only solutions I found to address the issues are: 
- only show tooltips for class `MuiTreeItem-label`, this stop the tooltip showing outside of the list elements
- dynamically switch the width between `auto` and `100%` for class `css-9l5vo-MuiCollapse-wrapperInner` at the width breakpoint of 280 for the out Box element

**Review by AronPetrauskas — APPROVED** [review]
Tested the code, I could not replicate the bugs. LGTM!

**Comment by syedsalehinipg** [comment]
Removed padding (24px) for TabPanel. Issue found when testing this component as part of the pre-release in VIRTO.
  
image

When trying it here, I noticed:
 
image

**Comment by syedsalehinipg** [comment]
The tests running on ubuntu are failing due to a [known issue](https://github.com/actions/runner-images/issues/7210) as it has been raised by other users on GitHub. It was previously fixed but seems to have started again as there are more comments referring to this today.

https://github.com/actions/runner-images/issues/7210

Will be keeping an eye on the issue for any updates.

File: package.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
     "start": "microbundle-crl watch --no-compress --format modern,cjs --jsxFragment React.Fragment",
     "storybook": "storybook dev -p 6006",
     "storybook:build": "storybook build",
-    "storybook:serve:start": "pm2 delete storybook -s 2> /dev/null || true && pm2 start ./serve-storybook.js --name storybook",
+    "storybook:serve:start": "pm2 start ./serve-storybook.js --name storybook",
```
Comment (syedsalehinipg) [inline package.json:20]: Adjusted script to not attempt to delete storybook on `storybook:serve:start`. The `storybook:serve:stop` script will delete storybook.

The`test:playwright` script should build, start, test then stop, so we shouldn't need to run the start and stop scripts separately.

**Pull request merged**